src/Compilers/Core/Portable/Collections/ImmutableArrayDictionary`2+KeyCollection.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    partial struct ImmutableArrayDictionary<TKey, TValue>
+    {
+        public readonly partial struct KeyCollection : IReadOnlyCollection<TKey>, ICollection<TKey>, ICollection
+        {
+            private readonly ImmutableArrayDictionary<TKey, TValue> _dictionary;
+
+            internal KeyCollection(ImmutableArrayDictionary<TKey, TValue> dictionary)
+            {
+                _dictionary = dictionary;
+            }
+
+            public int Count => _dictionary.Count;
+
+            bool ICollection<TKey>.IsReadOnly => true;
+
+            bool ICollection.IsSynchronized => true;
+
+            object ICollection.SyncRoot => ((ICollection)_dictionary).SyncRoot;
+
+            public Enumerator GetEnumerator()
+                => new Enumerator(_dictionary.GetEnumerator());
```

Comment:
Maybe we could return `_dictionary.Keys.GetEnumerator()` and not introduce the `KeyCollection.Enumerator` type.

Reply:
➡️ Covered by duplicate comment https://github.com/dotnet/roslyn/pull/47641#discussion_r487359513